### PR TITLE
feat(beer-encyclopedia): add data models and initial schema migration

### LINKED ISSUE #544: feat(beer-encyclopedia): design data models for breweries, beers, and provenance
Part of #541.

## Description

Design and implement the full encyclopedia schema — 10 tables for maximum richness and extensibility: styles, breweries, beers, ingredients, beer_ingredients, tasting_profiles, media, sources, entity_sources (provenance), community_corrections.

## Acceptance Criteria

- [ ] All 10 SQLAlchemy models created in `db/models/`
- [ ] First Alembic migration (`001_initial_schema.py`) creates all tables + indexes + pg_trgm extension
- [ ] GIN trigram indexes on `breweries.name`, `beers.name`
- [ ] `scripts/seed_styles.py` seeds ~15 styles
- [ ] Model tests pass (creation, relationships, constraints)
- [ ] `alembic upgrade head` applies cleanly

## Priority

Must-Have

## Estimation

L (5 SP)

## Sprint

Backlog (not yet planned)

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Step 3 of 6 for epic #541. Implements the full encyclopedia schema on top of the PostgreSQL infrastructure merged in #543. No API endpoints yet — those land in #545/#546.

- **10 ORM models** in `db/models/`: `Style`, `Brewery`, `Beer`, `Ingredient` + `BeerIngredient` junction, `TastingProfile` (1-to-1 with Beer), `Media` (polymorphic: exactly one of `beer_id`/`brewery_id` required via CHECK), `Source` + `EntitySource` (provenance tracking with JSONB `raw_data`, JSON fallback on SQLite), `CommunityCorrection` (moderation queue)
- **Hand-written migration** `001_initial_schema.py`:
  - 10 tables with FK cascades (`ON DELETE SET NULL` for `beers.brewery_id`/`beers.style_id` so historical records survive brewery deletion; `CASCADE` for owned rows)
  - CHECK constraints: ABV ranges on styles, 1–5 bounds on tasting scales, media parent requirement
  - B-tree indexes on common filter paths (city, country, brewery_type, FK columns, abv, moderation queue)
  - PostgreSQL-only DDL guarded by `_is_postgres()`: `CREATE EXTENSION pg_trgm` + GIN trigram indexes on `breweries.name` and `beers.name` — ready for the search endpoint in #546
- **Style seeder** `scripts/seed_styles.py`: idempotent upsert of 15 styles (the 8 slugs recognized by `ml/extract.py` + 7 mainstream additions: porter, pilsner, hefeweizen, dubbel, quadrupel, barleywine, blonde ale)
- **14 new tests** covering relationships, cascades, unique + check constraints, polymorphic parents, provenance triplet uniqueness, and seeder idempotence

## Design notes

- The `_is_postgres()` guard in the migration keeps the SQLite path (used in CI and local tests) free of PG-specific DDL without maintaining two migrations.
- `EntitySource.raw_data` uses `JSON().with_variant(JSONB(), "postgresql")` so the metadata is portable.
- The `Style.slug` values are aligned with `ml/extract.py` so the scan pipeline can resolve its string output to a persisted row without a mapping table.

## Checklist

- [x] Happy path + sad path + edge cases covered
- [x] `tsc --noEmit` passes (N/A — Python package)
- [x] Build passes (`alembic upgrade head` + `downgrade base` clean on SQLite)
- [x] Existing tests still green (16 existing + 14 new = 30/30)
- [x] No \`any\`, no inline styles introduced

Closes #544

### CODE REVIEW

File: packages/beer-encyclopedia/migrations/versions/001_initial_schema.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        sa.Column("alt_text", sa.String(length=255), nullable=True),
+        sa.Column("is_primary", sa.Boolean(), nullable=False, server_default=sa.false()),
+        sa.Column("uploaded_by", sa.String(length=100), nullable=True),
+        sa.Column("created_at", sa.DateTime(timezone=True), server_default=sa.func.now(), nullable=False),
+        sa.Column("updated_at", sa.DateTime(timezone=True), server_default=sa.func.now(), nullable=False),
+        sa.PrimaryKeyConstraint("id"),
+        sa.ForeignKeyConstraint(["beer_id"], ["beers.id"], ondelete="CASCADE"),
+        sa.ForeignKeyConstraint(["brewery_id"], ["breweries.id"], ondelete="CASCADE"),
+        sa.CheckConstraint(
+            "beer_id IS NOT NULL OR brewery_id IS NOT NULL",
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/beer-encyclopedia/migrations/versions/001_initial_schema.py]: Addressed in 48dff57. You are right — the OR constraint allowed (beer_id, brewery_id) to both be set, which breaks the polymorphic single-owner model. Changed to strict XOR in both the ORM and the migration:

```sql
(beer_id IS NOT NULL AND brewery_id IS NULL) OR
(beer_id IS NULL AND brewery_id IS NOT NULL)
```

Portable between PostgreSQL and SQLite (no need for `num_nonnulls`). Added regression tests covering both edge cases: `test_media_rejects_both_parents` and `test_media_accepts_single_parent`.

File: packages/beer-encyclopedia/db/models/beer.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        cascade="all, delete-orphan",
+    )
+    ingredients: Mapped[list[BeerIngredient]] = relationship(
+        back_populates="beer",
+        cascade="all, delete-orphan",
+    )
+    media: Mapped[list[Media]] = relationship(
+        back_populates="beer",
+        primaryjoin="Beer.id == Media.beer_id",
+    )
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/beer-encyclopedia/db/models/beer.py:64]: Addressed in 48dff57. Excellent catch — this would have been silently broken once XOR lands in #1/#3/#4, because SQLAlchemy would null media.beer_id and leave a (NULL, NULL) row that violates the new constraint.

Fix on `Beer.media` (and symmetrically `Brewery.media`):

```python
media: Mapped[list[Media]] = relationship(
    back_populates="beer",
    primaryjoin="Beer.id == Media.beer_id",
    cascade="all, delete-orphan",
    passive_deletes=True,
)
```

This matches the pattern already used on `tasting_profile` and `ingredients`, delegates the physical DELETE to the FK's `ON DELETE CASCADE` (no extra round-trips), and keeps the session state coherent.

Side effect I noticed while writing the regression tests: our SQLite test fixture did **not** enable `PRAGMA foreign_keys=ON`, so all previous cascade tests were passing for the wrong reason (ORM nulling FKs, not SQL cascades). The conftest now enables the pragma per engine via an event listener, aligning test behavior with production PostgreSQL.

File: packages/beer-encyclopedia/db/models/media.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    Exactly one of ``beer_id`` / ``brewery_id`` must be set (enforced by a
+    CHECK constraint). Polymorphic parent avoids a separate media table per
+    entity type while keeping referential integrity.
+    """
+
+    __tablename__ = "media"
+    __table_args__ = (
+        CheckConstraint(
+            "beer_id IS NOT NULL OR brewery_id IS NOT NULL",
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/beer-encyclopedia/db/models/media.py]: Addressed in 48dff57 — same fix as Codex's matching comment on the migration (applied XOR in both the ORM model and the migration to keep them in sync). See the full rationale on the first reply on this PR.

File: packages/beer-encyclopedia/migrations/versions/001_initial_schema.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/beer-encyclopedia/migrations/versions/001_initial_schema.py]: Addressed in 48dff57 — your suggested XOR expression was applied verbatim in both the migration and the ORM model. Two new tests cover the now-rejected (both set) case and the accepted (exactly one parent) case.

File: packages/beer-encyclopedia/migrations/versions/001_initial_schema.py [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    op.create_table(
+        "community_corrections",
+        sa.Column("id", sa.Uuid(), nullable=False),
+        sa.Column("entity_type", sa.String(length=30), nullable=False),
+        sa.Column("entity_id", sa.Uuid(), nullable=False),
+        sa.Column("field_name", sa.String(length=100), nullable=False),
+        sa.Column("old_value", sa.Text(), nullable=True),
+        sa.Column("new_value", sa.Text(), nullable=False),
+        sa.Column("reason", sa.Text(), nullable=True),
+        sa.Column("status", sa.String(length=20), nullable=False, server_default="pending"),
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/beer-encyclopedia/migrations/versions/001_initial_schema.py]: Addressed in 48dff57. I verified the actual SQL generated by SQLAlchemy 2.0 on both dialects — `server_default="pending"` is auto-quoted to `DEFAULT 'pending'` on both PostgreSQL and SQLite, so the bug you describe does not manifest in our version. However I still applied your suggestion (`sa.text("'pending'")`) for three reasons:

1. **Consistency** — the same migration uses `sa.func.now()`, `sa.false()`, `sa.true()` for other server defaults. Using a raw string here was the odd one out.
2. **Explicit beats implicit** — no reliance on an auto-quoting heuristic that could change between SA versions.
3. **Future-proof** — our code stays correct even if SQLAlchemy tightens its policy in a major release.

**Review by benoit-bremaud — COMMENTED** [review]
Self-approving after a full review cycle.

Delivered: the 10-table encyclopedia schema (styles, breweries, beers,
ingredients + beer_ingredients, tasting_profiles, media, sources +
entity_sources, community_corrections), a hand-written initial Alembic
migration with PostgreSQL-only DDL guarded by `_is_postgres()`
(pg_trgm extension + GIN trigram indexes on `breweries.name` and
`beers.name`), and an idempotent style seeder covering the 8 slugs
recognized by `ml/extract.py` plus 7 mainstream additions.

Review cycle addressed 5 comments (2 Codex P1 + 3 Copilot):

- Media parent ownership tightened from OR to strict XOR — both in the
  ORM model and the migration. New tests cover the previously-allowed
  (both set) case and the still-accepted (exactly one parent) case.
- Cascade added on `Beer.media` and `Brewery.media`
  (`cascade="all, delete-orphan", passive_deletes=True`) so deleting a
  beer/brewery with attached media delegates to the FK's
  `ON DELETE CASCADE` instead of nulling FKs and violating the new XOR.
- `community_corrections.status` server default switched from
  `"pending"` to `sa.text("'pending'")` for consistency with the
  migration's other defaults (`sa.func.now()`, `sa.false()`,
  `sa.true()`) and to remove the implicit reliance on SQLAlchemy
  auto-quoting.

Side-effect bonus uncovered while writing the cascade regression tests:
the SQLite test fixture wasn't activating `PRAGMA foreign_keys=ON`, so
prior cascade tests were passing for the wrong reason (ORM nulling FKs
rather than SQL cascades). The conftest now enables the pragma per
engine, aligning test behavior with production PostgreSQL.

Verifications: ruff clean, 34/34 tests pass (24 db tests + 10 existing),
`alembic upgrade` + `downgrade` clean on SQLite, SQL generation verified
on the PostgreSQL dialect for the `DEFAULT 'pending'` change.

No API endpoints yet — those land in #545 and #546.
